spec/horoshop/authorization_spec.rb

```diff
@@ -3,54 +3,53 @@
 require 'spec_helper'
 
 describe Horoshop::Authorization do
-  let(:instance) { double('Instance', username: 'user', password: 'pass', url: 'http://api.horosop.com') }
+  subject(:auth) { described_class.new(instance) }
+
+  let(:instance) { instance_double('Instance', username: 'user123', password: 'pass123', url: 'http://api.horosop.com') }
 
   before do
     allow(instance).to receive(:token=)
     allow(instance).to receive(:expiration_timestamp=)
-    allow(Time).to receive(:now).and_return(Time.now)
+    allow(Time).to receive(:now).and_return(Time.parse('2024-02-21 12:42:59 +0200'))
   end
 
-  subject(:authorization) { described_class.new(instance) }
-
   describe '#authorize' do
     context 'when authorization is successful' do
       let(:successful_response) do
         {
-          body: { 'status' => 'OK', 'response' => { 'token' => 'valid_token' } }.to_json,
+          body: { 'status' => 'OK', 'response' => { 'token' => '123qwe4' } }.to_json,
           headers: { 'Content-Type' => 'application/json' }
         }
       end
 
       before do
         stub_request(:post, 'http://api.horosop.com/api/auth/')
-          .with(body: { login: 'user', password: 'pass' })
+          .with(body: { login: 'user123', password: 'pass123' })
           .to_return(successful_response)
       end
 
       it 'sets the token and expiration_timestamp on the instance' do
-        authorization.authorize
-        expect(instance).to have_received(:token=).with('valid_token')
-        expect(instance).to have_received(:expiration_timestamp=).with(an_instance_of(Time))
+        expect(instance).to receive(:token=).with('123qwe4')
+        expect(instance).to receive(:expiration_timestamp=).with(Time.now + 600)
```

Comment:
Some time, when specs count increased this expect will produce error. 

You should expect `Time.parse('2024-02-21 12:42:59 +0200') + 600`

Reply:
stub Time.now used for any executions in described class and his child nodes, but not in spec. In the spec Time.now produced doing command as is

Reply:
understood, it means I misunderstood the execution of the before block